### ISSUE TITLE
`uninit` analysis misses uninitialized variables accessed through the pointers after integer  round-trip

### ISSUE THREAD

**Issue by jprotopopov-ut** [goblint/analyzer#2136]
Tested with nightly Goblint image (sha256:691055f2e25b2b101041a8d348c3fedc9da6143d5831e8aa2bad81d16f35aa86), the following test case

```c
// PARAM: --set ana.activated[+] uninit

#include <stdint.h>

int main(void) {
  int value;
  uintptr_t address = (uintptr_t)(void *)&value;
  int *pointer = (void *)address;
  return *pointer; // TODO WARN: uninitialized value
}
```

does not produce a warning on `*pointer` despite the fact that `pointer` is `&value` after round-trip through `uintptr_t`. Instead, Goblint reports `[Info][Unsound] Access to unknown address could be global`. However, with `conf/examples/very-precise.json` no warnings are reported at all, which is unsound.

The issue was discovered by a coding agent when testing the effects implementation

**Comment by jprotopopov-ut** [comment]
Update to the above: in further tests, I was able to get `Access to unknown address could be global` in very precise configuration too, after I reordered the command line arguments. Nevertheless, the general issue of `uninit` ignorance of the unknown pointer values still persists

**Comment by michael-schwarz** [comment]
The uninit analysis is largely unmainted. Do we want to revive it?
